### PR TITLE
[RFXCOM] Resolved PMD/checkstyle warnings

### PR DESCRIPTION
**Pull request by martinvw**
Resolved PMD/checkstyle warnings

See: https://community.openhab.org/t/static-code-analysis-tool/24489

Signed-off-by: Martin van Wingerden <martinvw@mtin.nl>

### CODE REVIEW

File: addons/binding/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/handler/RFXComHandler.java [hunk 1]
```diff
@@ -45,14 +44,11 @@
  * @author Pauli Anttila - Initial contribution
  */
 public class RFXComHandler extends BaseThingHandler implements DeviceMessageListener {
+    private final static int LOW_BATTERY_LEVEL = 1;
 
-    private Logger logger = LoggerFactory.getLogger(RFXComHandler.class);
+    private static final Logger LOGGER = LoggerFactory.getLogger(RFXComHandler.class);
```
Comment (kaikreuzer) [inline addons/binding/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/handler/RFXComHandler.java]: was this actually a check of the analysis tool? If so, we need to adapt it - our coding guidelines state that non-static loggers should be used whenever possible.

Comment (martinvw) [inline addons/binding/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/handler/RFXComHandler.java]: No but it was surrounded by `private static final Logger logger` declarations which I changed to `private static final Logger LOGGER`. 

And I (and most companies I worked) actually use and prefer the static ones, what are the/your objections against it.

Comment (kaikreuzer) [inline addons/binding/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/handler/RFXComHandler.java]: We had long discussions before defining [E.1 in our guidelines](https://www.eclipse.org/smarthome/documentation/development/guidelines.html). The main argument for the decision was the "IOC friendlyness" as we have to consider OSGI DS as an IOC mechanism. So please adapt the code again to match these guidelines - thanks!

Comment (martinvw) [inline addons/binding/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/handler/RFXComHandler.java]: If it's (that) important we should add our own rule to tackle this :-)

**Comment by martinvw** [comment]
I resolved them, we should consider adding a custom rule to check for it.

Because it's now still a point which requires manual checking and using static logger's is quite common for other projects so a lot of binding do contain this problem now and a lot of PR's most likely too.

I count after resolving mine:
- 78 in openhab 
- 42 in the smarthome repo.

**Comment by kaikreuzer** [comment]
> we should consider adding a custom rule to check for it.

Absolutely. As you can [see here](https://github.com/openhab/static-code-analysis/#e-logging), this is already "work in progress".

**Comment by martinvw** [comment]
Great, maybe if I have time I can took a look at one of the open issues

IMHO everything for this PR should now be done.

PS: does the mentionbot have an off-day today?

**Comment by kaikreuzer** [comment]
Perfect, thanks!

> PS: does the mentionbot have an off-day today?

Haha, no idea. Maybe too late at night... But anyhow, even without being mentioned by a bot, I will merge it :-)

**Pull request merged**